bridge/tracker/rootchain.go

```diff
@@ -0,0 +1,231 @@
+	"github.com/0xPolygon/polygon-edge/types"
+	"github.com/hashicorp/go-hclog"
+	"github.com/umbracle/go-web3"
+	"github.com/umbracle/go-web3/abi"
+	client "github.com/umbracle/go-web3/jsonrpc"
+	"math/big"
+	"os"
+	"path/filepath"
+)
+
+const (
+	//	required block depth for fetching events on the rootchain
+	DefaultBlockConfirmations = 6
+)
+
+//	contractABI is used to create query filter
+//	that matches events defined in a smart contract.
+type contractABI struct {
+	//	address of smart contract
+	address web3.Address
+
+	// signatures of events defined in smart contract
+	events []*abi.Event
+}
+
+//	eventIDs returns all the event signatures (IDs)
+//	defined in the smart contract.
+func (c *contractABI) eventIDs() (IDs []*web3.Hash) {
+	for _, ev := range c.events {
+		id := ev.ID()
+		IDs = append(IDs, &id)
+	}
+
+	return
+}
+
+//	loadABIs parses contracts from raw map.
+func loadABIs(abisRaw map[types.Address][]string) (contracts []*contractABI) {
+	for address, events := range abisRaw {
+		//	set smart contract address
+		contract := &contractABI{
+			address: web3.HexToAddress(address.String()),
+		}
+
+		//	set each event (defined in contract)
+		for _, ev := range events {
+			contract.events = append(contract.events, abi.MustNewEvent(ev))
+		}
+
+		//	append result
+		contracts = append(contracts, contract)
+	}
+
+	return
+}
+
+//	setupQueryFilter creates a log filter for the desired
+//	block range. Filter matches events defined in rootchain.go.
+func setupQueryFilter(from, to *big.Int, contracts []*contractABI) *web3.LogFilter {
+	queryFilter := &web3.LogFilter{}
+
+	//	set range of blocks to query
+	queryFilter.SetFromUint64(from.Uint64())
+	queryFilter.SetToUint64(to.Uint64())
+
+	//	set contract addresses and topics
+	for _, contract := range contracts {
+		//	append address
+		queryFilter.Address = append(queryFilter.Address, contract.address)
+
+		//	topics from all contracts must be in Topics[0]
+		queryFilter.Topics = append(queryFilter.Topics, contract.eventIDs())
+	}
+
+	return queryFilter
+}
+
+/* Header types parsed by the client */
+
+//	Ethereum header
+//type ethHeader struct {
+//	Difficulty   string        `json:"difficulty"`
+//	ExtraData    string        `json:"extraData"`
+//	GasLimit     string        `json:"gasLimit"`
+//	GasUsed      string        `json:"gasUsed"`
+//	LogsBloom    types.Bloom   `json:"logsBloom"`
+//	Miner        types.Address `json:"miner"`
+//	Nonce        string        `json:"nonce"`
+//	Number       string        `json:"number"`
+//	ParentHash   types.Hash    `json:"parentHash"`
+//	ReceiptsRoot types.Hash    `json:"receiptsRoot"`
+//	Sha3Uncles   types.Hash    `json:"sha3Uncles"`
+//	StateRoot    types.Hash    `json:"stateRoot"`
+//	Timestamp    string        `json:"timestamp"`
+//	TxRoot       types.Hash    `json:"transactionsRoot"`
+//	MixHash      types.Hash    `json:"mixHash"`
+//	Hash         types.Hash    `json:"hash"`
+//}
```

Comment:
Can you please explain in detail the issue with using Ethereum headers in the PR description? 🙏 

I know we discussed it internally, but it's better to have this sort of data available for easy reference

Reply:
Resolving because you added the description to the PR 👍